fvm/script.go

```diff
@@ -113,6 +113,22 @@ func (proc *ScriptProcedure) ShouldDisableMemoryAndInteractionLimits(
 	return ctx.DisableMemoryAndInteractionLimits
 }
 
+func (proc *ScriptProcedure) IsSnapshotReadTransaction() bool {
+	return true
+}
+
+func (proc *ScriptProcedure) ShouldCommitPrograms() bool {
+	return true
```

Comment:
q: iirc Script is read-only? do we commit programs for Scripts for meters?

Reply:
scripts are read-only and don't modify source code.

Given the same source code, the parsed program will be identical regardless whether it was used by a script or a transaction.  Hence it's safe to commit the program here.

We're committing the program to speed up future transaction/script which uses the same program